lightning/src/chain/channelmonitor.rs

```diff
@@ -3602,83 +3602,116 @@ mod tests {
 		let txid = Txid::from_hex("56944c5d3f98413ef45cf54545538103cc9f298e0575820ad3591376e2e0f65d").unwrap();
 
 		// Justice tx with 1 to_holder, 2 revoked offered HTLCs, 1 revoked received HTLCs
-		let mut claim_tx = Transaction { version: 0, lock_time: 0, input: Vec::new(), output: Vec::new() };
-		for i in 0..4 {
-			claim_tx.input.push(TxIn {
-				previous_output: BitcoinOutPoint {
-					txid,
-					vout: i,
-				},
-				script_sig: Script::new(),
-				sequence: 0xfffffffd,
-				witness: Vec::new(),
-			});
-		}
-		claim_tx.output.push(TxOut {
-			script_pubkey: script_pubkey.clone(),
-			value: 0,
-		});
-		let base_weight = claim_tx.get_weight();
-		let inputs_weight = vec![WEIGHT_REVOKED_OUTPUT, WEIGHT_REVOKED_OFFERED_HTLC, WEIGHT_REVOKED_OFFERED_HTLC, WEIGHT_REVOKED_RECEIVED_HTLC];
-		let mut inputs_total_weight = 2; // count segwit flags
-		{
-			let mut sighash_parts = bip143::SigHashCache::new(&mut claim_tx);
-			for (idx, inp) in inputs_weight.iter().enumerate() {
-				sign_input!(sighash_parts, idx, 0, inp, sum_actual_sigs, false);
-				inputs_total_weight += inp;
+		macro_rules! justice_tx_with_revoked_htlcs {
```

Comment:
Macros are a bit of a pain in general, and I don't think one is needed here.  You can just do a `for opt_anchors in (false, true) {` loop around the body.

Reply:
fixed